plugin/evm/validators/state_test.go

```diff
@@ -0,0 +1,288 @@
+	require.Error(err)
+	// adding the same nodeID should fail
+	err = state.AddNewValidator(ids.GenerateTestID(), nodeID, uint64(startTime.Unix()), true)
+	require.Error(err)
+
+	// get uptime
+	upDuration, lastUpdated, err := state.GetUptime(nodeID)
+	require.NoError(err)
+	require.Equal(time.Duration(0), upDuration)
+	require.Equal(startTime.Unix(), lastUpdated.Unix())
+
+	// set uptime
+	newUpDuration := 2 * time.Minute
+	newLastUpdated := lastUpdated.Add(time.Hour)
+	require.NoError(state.SetUptime(nodeID, newUpDuration, newLastUpdated))
+	// get new uptime
+	upDuration, lastUpdated, err = state.GetUptime(nodeID)
+	require.NoError(err)
+	require.Equal(newUpDuration, upDuration)
+	require.Equal(newLastUpdated, lastUpdated)
+
+	// set status
+	require.NoError(state.SetStatus(vID, false))
+	// get status
+	status, err := state.GetStatus(vID)
+	require.NoError(err)
+	require.False(status)
+
+	// delete uptime
+	state.DeleteValidator(vID)
+
+	// get deleted uptime
+	_, _, err = state.GetUptime(nodeID)
+	require.ErrorIs(err, database.ErrNotFound)
+}
+
+func TestWriteValidator(t *testing.T) {
+	require := require.New(t)
+	db := memdb.New()
+	state, err := NewState(db)
+	require.NoError(err)
+	// write empty uptimes
+	require.NoError(state.WriteState())
+
+	// load uptime
+	nodeID := ids.GenerateTestNodeID()
+	vID := ids.GenerateTestID()
+	startTime := time.Now()
+	state.AddNewValidator(vID, nodeID, uint64(startTime.Unix()), true)
+
+	// write state, should reflect to DB
+	require.NoError(state.WriteState())
+	require.True(db.Has(vID[:]))
+
+	// set uptime
+	newUpDuration := 2 * time.Minute
+	newLastUpdated := startTime.Add(time.Hour)
+	require.NoError(state.SetUptime(nodeID, newUpDuration, newLastUpdated))
+	require.NoError(state.WriteState())
+
+	// refresh state, should load from DB
+	state, err = NewState(db)
+	require.NoError(err)
+
+	// get uptime
+	upDuration, lastUpdated, err := state.GetUptime(nodeID)
+	require.NoError(err)
+	require.Equal(newUpDuration, upDuration)
+	require.Equal(newLastUpdated.Unix(), lastUpdated.Unix())
+
+	// delete
+	state.DeleteValidator(vID)
+
+	// write state, should reflect to DB
+	require.NoError(state.WriteState())
+	require.False(db.Has(vID[:]))
+}
+
+func TestParseValidator(t *testing.T) {
+	testNodeID, err := ids.NodeIDFromString("NodeID-CaBYJ9kzHvrQFiYWowMkJGAQKGMJqZoat")
+	require.NoError(t, err)
+	type test struct {
+		name        string
+		bytes       []byte
+		expected    *validatorData
+		expectedErr error
+	}
+	tests := []test{
+		{
+			name:  "nil",
+			bytes: nil,
+			expected: &validatorData{
+				lastUpdated: time.Unix(0, 0),
+				startTime:   time.Unix(0, 0),
+			},
+			expectedErr: nil,
+		},
+		{
+			name:  "empty",
+			bytes: []byte{},
+			expected: &validatorData{
+				lastUpdated: time.Unix(0, 0),
+				startTime:   time.Unix(0, 0),
+			},
+			expectedErr: nil,
+		},
+		{
+			name: "valid",
+			bytes: []byte{
+				// codec version
+				0x00, 0x00,
```

Comment:
should we create then modify this bytes programmatically? (like cutting it short, or modifying the initial bytes)

Reply:
not sure I understood this, we have cases for `short byte len` and `invalid codec version`

Reply:
I mean should we generate the valid bytes using the codec or should we leave it hardcoded as is?